src/contracts/core/DelegationManager.sol

```diff
@@ -528,132 +560,174 @@ contract DelegationManager is
     }
 
     /**
-     * @dev commented-out param (middlewareTimesIndex) is the index in the operator that the staker who triggered the withdrawal was delegated to's middleware times array
-     * This param is intended to be passed on to the Slasher contract, but is unused in the M2 release of these contracts, and is thus commented-out.
+     * @dev This function "descales" the scaledShares according to the totalMagnitude at the time of withdrawal completion.
+     * This will apply any slashing that has occurred since the scaledShares were initially set in storage. 
+     * If receiveAsTokens is true, then these descaled shares will be withdrawn as tokens.
+     * If receiveAsTokens is false, then they will be rescaled according to the current operator the staker is delegated to, and added back to the operator's scaledShares.
      */
     function _completeQueuedWithdrawal(
         Withdrawal calldata withdrawal,
         IERC20[] calldata tokens,
-        uint256, /*middlewareTimesIndex*/
         bool receiveAsTokens
     ) internal {
         bytes32 withdrawalRoot = calculateWithdrawalRoot(withdrawal);
 
         require(
             pendingWithdrawals[withdrawalRoot], "DelegationManager._completeQueuedWithdrawal: action is not in queue"
         );
-
         require(
-            withdrawal.startBlock + minWithdrawalDelayBlocks <= block.number,
-            "DelegationManager._completeQueuedWithdrawal: minWithdrawalDelayBlocks period has not yet passed"
+            withdrawal.startTimestamp + minWithdrawalDelay <= block.number,
+            "DelegationManager._completeQueuedWithdrawal: minWithdrawalDelay period has not yet passed"
         );
-
         require(
             msg.sender == withdrawal.withdrawer,
             "DelegationManager._completeQueuedWithdrawal: only withdrawer can complete action"
         );
+        require(
+            tokens.length == withdrawal.strategies.length,
+            "DelegationManager._completeQueuedWithdrawal: input length mismatch"
+        );
+
+        // read delegated operator for scaling and adding shares back if needed
+        address currentOperator = delegatedTo[msg.sender];
+        // descale shares to get the "real" strategy share values of the withdrawal
+        uint256[] memory descaledShares = ShareScalingLib.descaleSharesAtTimestamp(
+            avsDirectory,
+            withdrawal.delegatedTo,
+            withdrawal.strategies,
+            withdrawal.scaledShares,
+            withdrawal.startTimestamp + avsDirectory.DEFAULT_ALLOCATION_DELAY()
```

Comment:
need the deallocation delay here?

Reply:
Yes, you need to read the totalMagnitude at time of completion, not the current totalMagnitude in the existing helper functions

Reply:
emphasis on the DEallocation

Reply:
brainfart https://github.com/Layr-Labs/eigenlayer-contracts/pull/695/commits/287cdceb9484cf4257bfcec2da74319bba547fe1#diff-d323417ded59a593f4ea410225a9167b24198c4d2c05cbc0fe9af47d34b88810R643